include/red_black_tree.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2019, DSA authors.
+ * All rights reserved.
+ * Authors: Allen
+ */
+
+// 红黑树
+#pragma once
+
+#include <memory>
+#include <queue>
+#include <vector>
+#include <optional>
+#include <iostream>
+
+#include <binary_search_tree.h>
+
+namespace dsa {
+
+template <typename K, typename V>
+class RedBlackTree : public BinarySearchTree<K, V> {
+public:
+    using Node = typename BinarySearchTree<K, V>::Node;
+    using BinarySearchTree<K, V>::_root;
+    using BinarySearchTree<K, V>::left_rotate;
+    using BinarySearchTree<K, V>::right_rotate;
+
+    RedBlackTree() {}
+
+    // 插入键值对
+    // 不允许插入重复 key
+    template <typename Key, typename Value>
+    bool insert(Key&& key, Value&& val) {
+        auto z = new Node(std::make_pair(std::forward<Key>(key), std::forward<Value>(val)));
```

Comment:
Is it better to use unique_ptr here?

I can not find any release for this memory. make sure there is no memory leak in dtor.